File: scripts/process_logo.js
const sharp = require('sharp');
const fs = require('fs');

const inputPath = process.argv[2];
const outputPath = process.argv[3];

if (!inputPath || !outputPath) {
    console.error("Usage: node process_logo.js <input> <output>");
    process.exit(1);
}

// Function to process image
async function processImage() {
    try {
        console.log(`Processing ${inputPath}...`);

        // Load image
        const image = sharp(inputPath);

        // Get metadata to ensure we have data
        const metadata = await image.metadata();

        // Threshold strategy: 
        // Convert to raw buffer, iterate pixels, set alpha to 0 if dark.
        const { data, info } = await image
            .ensureAlpha()
            .raw()
            .toBuffer({ resolveWithObject: true });

        const pixelArray = new Uint8ClampedArray(data);

        for (let i = 0; i < pixelArray.length; i += 4) {
            const r = pixelArray[i];
            const g = pixelArray[i + 1];
            const b = pixelArray[i + 2];

            // Check if dark (Threshold 100)
            if (r < 100 && g < 100 && b < 100) {
                pixelArray[i + 3] = 0; // Set Alpha to 0
            }
        }

        await sharp(pixelArray, {
            raw: {
                width: info.width,
                height: info.height,
                channels: 4
            }
        })
            .png()
            .toFile(outputPath);

        console.log(`Success! Saved to ${outputPath}`);

    } catch (error) {
        console.error("Error processing image:", error);
        process.exit(1);
    }
}

processImage();

File: scripts/process_logo.py
from PIL import Image
import sys

def remove_dark_background(input_path, output_path):
    print(f"Processing {input_path}...")
    try:
        img = Image.open(input_path).convert("RGBA")
        datas = img.getdata()
        
        new_data = []
        for item in datas:
            # item is (R, G, B, A)
            # Filter out dark pixels (background)
            # Keeping white/light pixels (the icon)
            # Threshold: if R, G, and B are all less than 100 (dark gray)
            if item[0] < 100 and item[1] < 100 and item[2] < 100:
                new_data.append((0, 0, 0, 0)) # Make transparent
            else:
                # Keep the pixel (it's likely part of the white icon)
                new_data.append(item)
                
        img.putdata(new_data)
        img.save(output_path, "PNG")
        print(f"Saved to {output_path}")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    if len(sys.argv) < 3:
        print("Usage: python process_logo.py <input> <output>")
    else:
        remove_dark_background(sys.argv[1], sys.argv[2])
